website/docs/developer/research/research-lambda-container.md

```diff
@@ -148,6 +148,152 @@ Based on this research, it looks like using lambda as containers is possible whe
+  assume_role_policy   = data.aws_iam_policy_document.assume_ecs_tasks_role_policy_document.json
+  permissions_boundary = var.permissions_boundary_arn
+  tags                 = var.tags
+}
+
+resource "aws_iam_role_policy" "sqs_task_role_policy" {
+  name   = "${var.prefix}_orca_sqs_task_role_policy"
+  role   = aws_iam_role.orca_ecs_tasks_role.id
+  policy = data.aws_iam_policy_document.sqs_task_policy_document.json
+}
+
+# This role is required by tasks to pull container images and publish container logs to Amazon CloudWatch on your behalf.
+resource "aws_iam_role" "orca_ecs_task_execution_role" {
+  name                 = "${var.prefix}_orca_ecs_task_execution_role"
+  assume_role_policy   = data.aws_iam_policy_document.assume_ecs_tasks_role_policy_document.json
+  permissions_boundary = var.permissions_boundary_arn
+  tags                 = var.tags
+}
+
+
+resource "aws_iam_role_policy_attachment" "ecs_role_policy" {
+  role       = aws_iam_role.orca_ecs_task_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/AdministratorAccess"
+}
+
+resource "aws_ecs_cluster" "test-cluster" {
+  name = "${var.prefix}_orca_ecs_cluster"
+  capacity_providers = ["FARGATE"]
+  default_capacity_provider_strategy {
+    capacity_provider = "FARGATE"
+    weight            = 100
+  }
+}
+
+resource "aws_ecs_task_definition" "task" {
+  family                   = "${var.prefix}_orca_sqs_task"
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = "4096"
+  memory                   = "8192"
+  task_role_arn            = aws_iam_role.orca_ecs_tasks_role.arn
+  execution_role_arn       = aws_iam_role.orca_ecs_task_execution_role.arn
+  container_definitions    = <<DEFINITION
+[
+  {
+    "name": "sqs_task_container",
+    "image": "236859827343.dkr.ecr.us-west-2.amazonaws.com/adorn-test-repo:latest",
+    "cpu": 4096,
+    "memory": 256,
+    "networkMode": "awsvpc",
+    "environment": [
+      {
+        "name": "TARGET_QUEUE_URL",
+        "value": "https://sqs.us-west-2.amazonaws.com/236859827343/doctest-orca-internal-report-queue.fifo"
+      }
+    ],
+    "logConfiguration": {
+      "logDriver": "awslogs",
+      "options": {
+        "awslogs-create-group": "true",
+        "awslogs-region": "us-west-2",
+        "awslogs-group": "${var.prefix}_orca_sqs_task",
+        "awslogs-stream-prefix": "ecs"
+      }
+    }
+  }
+]
+DEFINITION
+}
+:::note
+The above example was developed for deploying a simple task that posts to an sqs queue. Names and values should be changed to match new use cases.
+:::
+:::warning
+Applying admin permissions to orca_ecs_task_execution_role is likely overly permissive.
+:::
+- The task can now be run in the cluster.
+This can be done through [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.run_task) or the GUI,
+though the former is presently untested.
+
+
```

Comment:
Will it be possible to some info on cost analysis of using ECS instead of lambdas?

I know we "pay as we go" for lambdas which is cheap. But I am not so familiar with ECS cost. Does the cluster run always and if so, how much do we pay per month? 

I think the cost could be a concern.

Reply:
So long as we stick with Fargate tasks, we can make use of auto-scaling. Added link.

Reply:
sounds good.

What about the fargate cluster? Do you have to pay for that even when we are not using it?

Reply:
Fargate refers to a task run under a ECS cluster under the Fargate protocol, so the "fargate container" is just the ECS container. I'll add a couple clarifying details to the doc.